enhancements/rhcos/liveisoinstall.md

```diff
@@ -0,0 +1,95 @@
+---
+title: RHEL CoreOS Live ISO/installer
+authors:
+  - "@cgwalters"
+reviewers:
+  - "@coreos-team"
+approvers:
+  - "@coreos-team"
+creation-date: 2020-02-07
+last-updated: 2020-02-07
+status: implementable
+---
+
+# Rebase RHEL CoreOS 4.5 on Fedora CoreOS $latest
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RHEL CoreOS is built from RHEL components, but uses configuration and tooling from Fedora CoreOS.  They share many things such as the installer code, etc.
+
+This enhancement tracks rebasing RHCOS 4.5 on the latest FCOS code, in particular the installer.
+
+### New "Live" installer
+
+Currently, RHCOS 4.4 and below offer a kernel/initramfs and ISO that can run a shell script installation program from the initramfs.
+
+In FCOS $latest, there is a Live ISO/PXE media that runs the full OS (for example, it includes `podman`), and a [coreos-installer](https://github.com/coreos/coreos-installer/) program was rewritten.
+
+## Motivation
+
+This allows administrators to run arbitrary code before and after the installation process by providing Ignition that wraps the installer.  For example, higher level tooling like Foreman and Ironic can easily `curl` a URL to report provisioning success, or even pull down and launch a container before/after.
+
+An additional goal is to allow administrators to more easily run RHCOS as a "live" system for *interactive* experiementation and testing.  For example, an administrator could run the system from a USB stick to validate the network interface names, etc.
+
+### Goals
+
+A live ISO and new `coreos-installer` binary are shipped, tested and work in all scenarios supported by the existing CoreOS installer, and contain the new functionality.
+
+### Non-Goals
+
+Support for Live/diskless nodes joining a cluster.
+
+## Proposal
+
+Switch the RHCOS build process to generate the new Live ISO media.
+
+Add a basic (virtualized) test case for the CoreOS installer.
+
+Ship the new `coreos-install` binary at https://mirror.openshift.com and as an RPM in the OpenShift channel.
+
+Switch the `e2e-metal` job in Packet to use the new CoreOS installer.
+
+Update the documentation.
+
+Drop the old CoreOS installer (hopefully).
+
+### Risks and Mitigations
+
+It's unknown exactly how difficult it will be for customers who are automating OpenShift installations to do the conversion, in particular for customers who want to be able to install both 4.4 and 4.5 for example.
```

Comment:
The OpenShift bare metal IPI process relies on having pre-prepared disk images that can be written to the system without having to interact with an installer process. Does this enhancement propose eliminating that?

Reply:
No, we still have `-metal` images suitable for `dd`.

More broadly speaking, this is basically replacing the "dd in initramfs" with "dd in a full live OS".